src/main/java/edu/cornell/kfs/paymentworks/service/impl/PaymentWorksKfsServiceImpl.java

```diff
@@ -57,394 +52,314 @@
 
 	@Override
 	public boolean directVendorEdit(PaymentWorksVendor paymentWorksVendor) {
-
 		boolean success = true;
 		PaymentWorksVendorUpdateConversionUtil paymentWorksVendorUpdateConversionUtil = new PaymentWorksVendorUpdateConversionUtil();
-
 		String vendorNumberList = paymentWorksVendor.getVendorNumberList();
 		List<String> vendorNumbers = Arrays.asList(vendorNumberList.split("\\s*,\\s*"));
 
-		// loop through vendors
 		for (String vendorNumber : vendorNumbers) {
-
-			// retrieve vendor from kfs
 			VendorDetail vendorDetail = vendorService.getVendorDetail(vendorNumber);
-
 			if (ObjectUtils.isNotNull(vendorDetail)) {
 				if (!paymentWorksVendorUpdateConversionUtil.duplicateFieldsOnVendor(vendorDetail, paymentWorksVendor)) {
-					// set the necessary fields
-					vendorDetail = paymentWorksVendorUpdateConversionUtil.createVendorDetailForEdit(vendorDetail, null,
-							paymentWorksVendor);
-
-					// save to DB, no document
+					vendorDetail = paymentWorksVendorUpdateConversionUtil.createVendorDetailForEdit(vendorDetail, null, paymentWorksVendor);
 					try {
 						businessObjectService.save(vendorDetail);
 					} catch (Exception e) {
-						GlobalVariables.getMessageMap().putError("vendorDetail", KFSKeyConstants.ERROR_CUSTOM,
-								e.getCause().getMessage());
+						GlobalVariables.getMessageMap().putError("vendorDetail", KFSKeyConstants.ERROR_CUSTOM, e.getCause().getMessage());
 						success = false;
 						break;
 					}
 				} else {
-					GlobalVariables.getMessageMap().putError("vendorDetail", KFSKeyConstants.ERROR_CUSTOM,
-							"Duplicate vendor update detected");
+					GlobalVariables.getMessageMap().putError("vendorDetail", KFSKeyConstants.ERROR_CUSTOM, "Duplicate vendor update detected");
 					success = false;
 				}
 			}
 		}
-
 		return success;
 	}
-
+	
 	@Override
 	public boolean directAchEdit(PaymentWorksVendorUpdatesDTO vendorUpdate, String vendorNumberList) {
-
-		boolean isSuccessful = false;
-
+		boolean successful = false;
 		String routingNumber = getRoutingNumber(vendorUpdate.getField_changes().getField_changes());
 		String accountNumber = getAccountNumber(vendorUpdate.getField_changes().getField_changes());
-
-		// check if bank exists
-		if (StringUtils.isNotEmpty(routingNumber)) {
-			ACHBank achBank = achBankService.getByPrimaryId(routingNumber);
-
-			if (ObjectUtils.isNull(achBank)) {
-				GlobalVariables.getMessageMap().putError("payeeAchAccount", KFSKeyConstants.ERROR_CUSTOM,
-						"ACH Bank does not exist for routing number");
-				return isSuccessful;
-			}
-		}
-
-		// check if we do not have at least routing or account number
-		// if not, throw an error
-		if (StringUtils.isEmpty(routingNumber) && StringUtils.isEmpty(accountNumber)) {
-			GlobalVariables.getMessageMap().putError("payeeAchAccount", KFSKeyConstants.ERROR_CUSTOM,
-					"Routing number or account number required");
-			return isSuccessful;
+		
+		if (validateRoutingNumberAndAccountNumber(routingNumber, accountNumber)) {
```

Comment:
I agree, it does not need more cleaning up.  I think we will find out the right implementation when we turn  on the ach functionality.  That said, I did a compare to the original uconn code, and I missed a setting successful to true at the end of the for loop.  Also it looks like the flow seen here is what UConn did, so I think when we turn this on, we will want to improve some naming, and reorgnizae to make more easy to understand what is going on.  Sound OK?